clients/admin-ui/src/features/data-discovery-and-detection/action-center/hooks/useDiscoveredSystemAggregateColumns.tsx

```diff
@@ -32,161 +35,320 @@ interface UseDiscoveredSystemAggregateColumnsProps {
   onTabChange: (tab: ActionCenterTabHash) => Promise<void>;
   consentStatus?: ConsentAlertInfo;
   rowClickUrl?: (record: SystemStagedResourcesAggregateRecord) => string;
+  resourceType?: StagedResourceTypeValue;
 }
 
+interface ColumnBuilderParams {
+  monitorId: string;
+  readonly: boolean;
+  allowIgnore?: boolean;
+  onTabChange: (tab: ActionCenterTabHash) => Promise<void>;
+  consentStatus?: ConsentAlertInfo;
+  rowClickUrl?: (record: SystemStagedResourcesAggregateRecord) => string;
+  isDataUsesExpanded: boolean;
+  isLocationsExpanded: boolean;
+  isDomainsExpanded: boolean;
+  dataUsesVersion: number;
+  locationsVersion: number;
+  domainsVersion: number;
+  setIsDataUsesExpanded: (value: boolean) => void;
+  setIsLocationsExpanded: (value: boolean) => void;
+  setIsDomainsExpanded: (value: boolean) => void;
+  setDataUsesVersion: (fn: (prev: number) => number) => void;
+  setLocationsVersion: (fn: (prev: number) => number) => void;
+  setDomainsVersion: (fn: (prev: number) => number) => void;
+}
+
+const isIDP = (resourceType?: StagedResourceTypeValue): boolean =>
+  resourceType === StagedResourceTypeValue.OKTA_APP;
+
+const buildIdpColumns = ({
+  rowClickUrl,
+}: Pick<
+  ColumnBuilderParams,
+  "rowClickUrl"
+>): ColumnsType<SystemStagedResourcesAggregateRecord> => [
+  {
+    title: "App Name",
+    dataIndex: "name",
+    key: "app_name",
+    fixed: "left",
+    render: (_, record) => (
+      <DiscoveredSystemStatusCell system={record} rowClickUrl={rowClickUrl} />
+    ),
+  },
+  {
+    title: "Type",
+    dataIndex: "metadata",
+    key: "app_type",
+    render: (metadata: OktaAppMetadata) => (
+      <span>{metadata?.app_type || "-"}</span>
+    ),
+  },
+  {
+    title: "Status",
+    dataIndex: "metadata",
+    key: "status",
+    render: (metadata: OktaAppMetadata) => (
+      <span>{metadata?.status || "-"}</span>
+    ),
+  },
+  {
+    title: "Vendor",
+    dataIndex: "vendor_id",
+    key: "vendor",
+    render: (
+      vendorId: string,
+      record: SystemStagedResourcesAggregateRecord,
+    ) => (
+      <VendorMatchBadge
+        vendorName={vendorId}
+        vendorLogoUrl={record.metadata?.vendor_logo_url}
+        confidence={record.metadata?.vendor_match_confidence}
+      />
+    ),
+  },
+  {
+    title: "Sign-on URL",
+    dataIndex: "metadata",
+    key: "sign_on_url",
+    render: (metadata: OktaAppMetadata) =>
+      metadata?.sign_on_url ? (
+        <a
+          href={metadata.sign_on_url}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          View
+        </a>
+      ) : null,
+  },
+  {
+    title: "Created",
+    dataIndex: "metadata",
+    key: "created",
+    render: (metadata: OktaAppMetadata) => (
+      <span>
+        {metadata?.created
+          ? new Date(metadata.created).toLocaleDateString()
+          : "-"}
```

Comment:
nit: I feel like `"-"` should be a constant somewhere so we can see `NO_VALUE` instead, and centralize what that is... but that can be refactored in the future and is probably littered everywhere